modules/artifact/src/main/java/org/mule/module/artifact/classloader/IBMMQResourceReleaser.java

```diff
@@ -382,4 +391,51 @@ private void processThreadLocalMap(Field threadLocalMapTableField, Object thread
       }
     }
   }
+
+  /**
+   * Close JMSCCThreadPoolMaster threads pending to close to avoid thread leaks after redeployment
+   */
+  private void disposeMQThreads() {
+    List<Thread> threads = ThreadUtils.getAllThreads().stream()
```

Comment:
We are honouring [the property](https://github.com/mulesoft/mule-ibm-mq-connector/blob/f550d3a865f9375a8885d75560183b9d193047bf/src/main/java/com/mulesoft/connectors/ibmmq/internal/connection/IBMMQConnectionProvider.java#L281) to skip the disposal.